test/test_settings.py

```diff
@@ -367,125 +369,65 @@ class TestSettingsEquality(object):
         """
-        Two objects that are not equal to each other do not compare equal to
-        each other using the == operator.
+        An object compares equal to itself using the == operator and the !=
+        operator.
         """
-        a = self.an_instance()
-        b = self.another_instance()
-        assert not (a == b)
+        assert (settings == settings)
+        assert not (settings != settings)
 
-    def test_different_ne(self):
+    @given(settings=SettingsStrategy, o_settings=SettingsStrategy)
+    def test_equality_multiple(self, settings, o_settings):
         """
-        Two objects that are not equal to each other compare not equal to each
-        other using the != operator.
+        Two objects compare themselves using the == operator and the !=
+        operator.
         """
-        a = self.an_instance()
-        b = self.another_instance()
-        assert (a != b)
+        if settings == o_settings:
+            assert settings == o_settings
+            assert not (settings != o_settings)
+        else:
+            assert settings != o_settings
+            assert not (settings == o_settings)
 
-    def test_another_type_eq(self):
+    @given(settings=SettingsStrategy)
+    def test_another_type_equality(self, settings):
         """
         The object does not compare equal to an object of an unrelated type
         (which does not implement the comparison) using the == operator.
         """
-        a = self.an_instance()
-        b = object()
-        assert not (a == b)
-
-    def test_another_type_ne(self):
-        """
-        The object compares not equal to an object of an unrelated type (which
-        does not implement the comparison) using the != operator.
-        """
-        a = self.an_instance()
-        b = object()
-        assert (a != b)
+        obj = object()
+        assert (settings != obj)
+        assert not (settings == obj)
 
-    def test_delegated_eq(self):
+    @given(settings=SettingsStrategy)
+    def test_delegated_eq(self, settings):
         """
-        The result of comparison using == is delegated to the right-hand
-        operand if it is of an unrelated type.
+        The result of comparison is delegated to the right-hand operand if
+        it is of an unrelated type.
         """
         class Delegate(object):
             def __eq__(self, other):
                 return [self]
 
-        a = self.an_instance()
-        b = Delegate()
-        assert (a == b) == [b]
-
-    def test_delegate_ne(self):
-        """
-        The result of comparison using != is delegated to the right-hand
-        operand if it is of an unrelated type.
-        """
-        class Delegate(object):
             def __ne__(self, other):
                 return [self]
 
-        a = self.an_instance()
-        b = Delegate()
-        assert (a != b) == [b]
+        delg = Delegate()
+        assert (settings == delg) == [delg]
+        assert (settings != delg) == [delg]
```

Comment:
I realise we nicked this test from elsewhere, but would it be better to tweak the return values from `__eq__` and `__ne__` so we can be sure we're really calling the right method?

e.g.

```python
def __eq__(self, other):
    return [self]

def __ne__(self, other):
    return [self, self]
```